lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -178,6 +182,7 @@ def get_child_resources(request, catalog, resource, file)
       end
       replace_metadata(request, children[meta.relative_path], meta)
     end
+    replace_metadata(request, resource, parent_meta)
```

Comment:
As far as I can tell File.perform_recursion() always includes the top level (.relative_path = '.'), so technically there should always a point in the loop where parent_meta is set.